webapp/packages/core-blocks/src/useFuzzySearch.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import MiniSearch, { type Options } from 'minisearch';
+import { useEffect, useRef } from 'react';
+
+interface UseSearchProps<T> extends Options {
+  sourceProposals: T[];
+}
+
+const DEFAULT_THRESHOLD = 0.4;
+
+function createSearchInstance<T>(options: Options) {
+  return new MiniSearch<T>(options);
+}
+
+export function useFuzzySearch<T extends object>({ sourceProposals, ...options }: UseSearchProps<T>) {
+  const minisearchOptions = { ...options };
```

Comment:
in case this hook is specific binding of minisearch for react, then we probably want to keep original interface as much as possible